### ISSUE TITLE
[rush-lib] Expose more interfaces...

### ISSUE THREAD

**Issue by dfee** [microsoft/rushstack#1870]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [x] Feature
- [ ] Bug

**Please describe the actual behavior.**
You don't export the awesome interfaces you've developed, so that I need to do the following:
```ts
import { IRushConfigurationProjectJson } from "@microsoft/rush-lib/lib/api/RushConfigurationProject";
```

In my quest for automation as I build on-top of Rush, this is key.

**Comment by halfnibble** [comment]
@dfee I agree these interfaces should be exported. Since you're already familiar with where these interfaces are located, and which ones are useful, could you make a PR exporting them? It would be greatly appreciated. Thanks!

**Comment by dfee** [comment]
Just saw this. More than happy to.

Before I do that, are there any other interfaces that come to mind that ought to be exported?

**Comment by dfee** [comment]
@halfnibble just went ahead and created it. LMK

**Comment by dfee** [comment]
There are a few more missing exports... but adding them creates significantly more downstream additions of missing AE exports.
